nameko/rpc.py

```diff
@@ -561,12 +621,31 @@ def call_async(self, *args, **kwargs):
         rpc_call = self._call(*args, **kwargs)
         return rpc_call
 
+    def with_options(self, **options):
+        """Create a new client with modified options."""
+        extra_keys = set(options).difference(self.updatable_options)
+        if extra_keys:
+            raise ValueError(
+                'Given options {} are not available in client update.'
+                .format(extra_keys))
+        return self._clone(**options)
+
     def _call(self, *args, **kwargs):
         if not self.fully_specified:
             raise ValueError(
                 "Cannot call unspecified method {}".format(self.identifier)
             )
 
+        if (
+                self.expiration is not None and
+                self.timeout is None
+        ):
+            # TODO: Figure out how to check this if "expiration" is set on the
+            # publisher itself, which we don't have access to here.
```

Comment:
Yes this was the kind of thing I was concerned about.
However, in the mean time I think this case might be covered:

The publisher option "expiration" is also passed to the client, so in this case while the client could not inspect the publisher, the client itself would actually have the expiration option set to the same value as the publisher and would trigger this `ValueError`.

While this isn't 100% "safe" since someone could later change the code without realizing this can cause problems, I think it's probably good enough if it works, but I will make extra sure these cases are tested (which would also guard against changes) and change the comment appropriately if that sounds ok?